iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/bucket/AffiliationMainSectionBucketHasher.java

```diff
@@ -0,0 +1,83 @@
+package eu.dnetlib.iis.wf.affmatching.bucket;
+
+import java.util.List;
+
+import eu.dnetlib.iis.wf.affmatching.model.AffMatchAffiliation;
+import eu.dnetlib.iis.wf.affmatching.model.AffMatchOrganization;
+import eu.dnetlib.iis.wf.affmatching.orgsection.OrganizationSection;
+import eu.dnetlib.iis.wf.affmatching.orgsection.OrganizationSection.OrgSectionType;
+import eu.dnetlib.iis.wf.affmatching.orgsection.OrganizationSectionsSplitter;
+
+/**
+ * An implementation of {@link BucketHasher} that hashes {@link AffMatchOrganization}.
+ * Hash is generated from 'main' (most significant) section of {@link AffMatchOrganization#getName()}.
+ * 
+ * @author madryk
+ */
+public class AffiliationMainSectionBucketHasher implements BucketHasher<AffMatchAffiliation> {
```

Comment:
Code duplication with `OrganizationMainSectionBucketHasher`. Probably good idea to extract common code to some `MainSectionBucketHasher`.